packages/roosterjs-plugin-react/lib/util/removeReactTagsOnExtractContent.ts

```diff
@@ -0,0 +1,19 @@
+import {
+    REACT_COMPONENT_DATA_KEY,
+    REACT_COMPONENT_INSTANCE_ID,
+    REACT_COMPONENT_SHARABLE_STATE,
+} from './constants';
+
+const simpleReactAttributes = new RegExp(
+    `(${REACT_COMPONENT_DATA_KEY}|${REACT_COMPONENT_INSTANCE_ID}|${REACT_COMPONENT_SHARABLE_STATE})=("[^"]*"|'[^']*'|[^\t\n\r\f ><"'\`]*)`,
+    'gim'
+);
+
+export default function removeReactTagsOnExtractContent(html: string): string {
+    if (html) {
+        const withoutSimpleTags = html.replace(simpleReactAttributes, '');
```

Comment:
We should only filter these strings within an HTML tag, not the content

Reply:
Sounds good.

Should I hold with this string scanning approach, or do you think I ought to parse the string to a DOM? I was a little worried about the perf impact of parsing & re-serializing the HTML string

Reply:
I think we can combine the regex search and simple string search to optimize for perf.

Here's what I did for image proxy, it firstly match with a regex, then check if the match result is inside a tag or not:

            bool replaced = false;
            char[] charToSearch = new char[] { '<', '>' };

            content = WebImageController.scriptRegex.Replace(
                content,
                match =>
                {
                    replaced = true;
                    return string.Empty;
                });

            content = WebImageController.onEventRegex.Replace(
                content,
                match =>  
                {
                    int c = content.LastIndexOfAny(charToSearch, match.Index - 1);
                
                    // Check if the match is inside a tag
                    if (content[c] == '<')
                    {
                        replaced = true;
                        return string.Empty;
                    }
                    else
                    {
                        return match.Value;
                    }
                });

            newContent = content;
            return replaced;